### PR TITLE
[ci] Add x64 specific Windows platform 

### PR DESCRIPTION
**Pull request by loic-sharma**
This begins to migrate the Windows CI tests to use a new x64 specific Windows platform. For now these duplicate the existing tests in `bringup` mode.

A subsequent pull request (https://github.com/flutter/packages/pull/5141) will complete the migration by removing the old Windows tests and removing the `bringup` from the new x64 specific tests. Here are the resulting changes from both steps of the migration: https://github.com/flutter/packages/compare/main...loic-sharma:flutter-packages:windows_x64_ci_finalize

In the future, a new platform will be added for Windows Arm64 tests.

Prepares for https://github.com/flutter/flutter/issues/129813

Subsequent pull requests:
* https://github.com/flutter/packages/pull/5141
* https://github.com/flutter/packages/pull/5142

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [relevant style guides] and ran the auto-formatter. (Unlike the flutter/flutter repo, the flutter/packages repo does use `dart format`.)
- [x] I signed the [CLA].
- [x] The title of the PR starts with the name of the package surrounded by square brackets, e.g. `[shared_preferences]`
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated `pubspec.yaml` with an appropriate new version according to the [pub versioning philosophy], or this PR is [exempt from version changes].
- [ ] I updated `CHANGELOG.md` to add a description of the change, [following repository CHANGELOG style].
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [ ] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/packages/blob/main/CONTRIBUTING.md
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[relevant style guides]: https://github.com/flutter/packages/blob/main/CONTRIBUTING.md#style
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat
[pub versioning philosophy]: https://dart.dev/tools/pub/versioning
[exempt from version changes]: https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#version-and-changelog-updates
[following repository CHANGELOG style]: https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#changelog-style
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests

### CODE REVIEW

File: .ci.yaml [hunk 1]
```diff
@@ -905,7 +906,7 @@ targets:
           {"dependency": "vs_build", "version": "version:vs2019"}
         ]
 
-  - name: Windows dart_unit_tests_shard_1 master
+  - name: Windows_x64 dart_unit_tests_shard_1 master
```
Comment (stuartmorgan) [inline .ci.yaml:909]: You can't just rename tasks unfortunately; ci.yaml doesn't allow for that workflow. You have to add a whole new duplicate set of tasks in bringup mode, and then do a second PR—after the first has both landed and propagated through the infra system—to remove the old ones and enable the new ones.

Comment (loic-sharma) [reply .ci.yaml:909]: Sounds good, will do!

File: .ci.yaml [hunk 2]
```diff
@@ -58,6 +58,15 @@ platform_properties:
...
       os: Windows
+  windows_x64:
+    properties:
+      dependencies: >
+        [
+          {"dependency": "certs", "version": "version:9563bb"}
+        ]
+      device_type: none
+      os: Windows
+      cpu: x86
```
Comment (loic-sharma) [inline .ci.yaml:69]: This line was added from the existing `windows` platform properties.

**Review by stuartmorgan — APPROVED** [review]
LGTM

I forgot we're still using VS 2019. We should probably look into getting that updated; I don't know if infra has 2022 available for us to depend on.

**Comment by loic-sharma** [comment]
@stuartmorgan Supporting VS2022 is tracked by https://github.com/flutter/flutter/issues/96138. FWIW, VS2019 will receive security updates until April 2029.

**Pull request merged**